app/models/conversation.rb

```diff
@@ -39,4 +40,26 @@ def self.share_conversation(user, match)
   def not_current_user(current_user)
     users.where.not(id: current_user.id).to_a
   end
+
+  def self.with_unread_messages_count_for(user_id)
+    Conversation
+      .select('conversations.*, conversation_users.unread_messages AS unread_count')
+      .joins(:conversation_users)
+      .where(conversation_users: { user_id: user_id })
+  end
+
+  def new_message(user_id)
+    conversations_with_new_messages =
+      conversation_users.where.not(user_id: user_id)
+    conversations_with_new_messages.each do |conversation_user|
+      conversation_user.update!(unread_messages:
```

Comment:
Since this is logic for the conversation_user, I would move this to that model. For example a method `increment_messages`.
Then you can call: `conversations_with_new_messages.each(&:increment_messages)`

Reply:
I had done it like this in the past. But then someone suggested I should do it on conversation side since the methods were proper from a conversation. I don't think ill change it back. But for next time I know :)